File: 3731-find-missing-elements/3731-find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        // Store the frequency of each number
        unordered_map<int, int> mp;

        int maxi = 0;
        int mini = 200;

        vector<int> ans;

        // Find the minimum and maximum element
        // while storing all numbers in the hash map
        for (int i : nums) {
            mp[i]++;
            maxi = max(maxi, i);
            mini = min(mini, i);
        }

        // Check every number in the range [mini, maxi]
        // If a number is not present in the map,
        // it is a missing element
        for (int i = mini; i <= maxi; i++) {
            if (mp.find(i) == mp.end())
                ans.push_back(i);
        }

        return ans;
    }
};

File: 3731-find-missing-elements/3731-find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        # Dictionary to store all the elements present in the array
        mp = {}

        mini = 200
        maxi = 0
        ans = []

        # Find the minimum and maximum element
        # while marking every number as present
        for i in nums:
            mp[i] = 1
            maxi = max(maxi, i)
            mini = min(mini, i)

        # Traverse the range [mini, maxi]
        # Any number not present in the dictionary is missing
        for i in range(mini, maxi + 1):
            if i not in mp:
                ans.append(i)

        return ans